# mod/rng-parse/mod.xml lines 16-17 - 's/java.io.IOException/EOFException/'

--- a/gensrc/main/com/thaiopensource/relaxng/parse/compact/CompactSyntaxTokenManager.java
+++ b/gensrc/main/com/thaiopensource/relaxng/parse/compact/CompactSyntaxTokenManager.java
@@ -219,7 +219,7 @@ private int jjMoveStringLiteralDfa0_0(){
 }
 private int jjMoveStringLiteralDfa1_0(long active0){
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(0, active0);
       return 1;
    }
@@ -264,7 +264,7 @@ private int jjMoveStringLiteralDfa2_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(0, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(1, active0);
       return 2;
    }
@@ -307,7 +307,7 @@ private int jjMoveStringLiteralDfa3_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(1, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(2, active0);
       return 3;
    }
@@ -342,7 +342,7 @@ private int jjMoveStringLiteralDfa4_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(2, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(3, active0);
       return 4;
    }
@@ -387,7 +387,7 @@ private int jjMoveStringLiteralDfa5_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(3, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(4, active0);
       return 5;
    }
@@ -426,7 +426,7 @@ private int jjMoveStringLiteralDfa6_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(4, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(5, active0);
       return 6;
    }
@@ -465,7 +465,7 @@ private int jjMoveStringLiteralDfa7_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(5, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(6, active0);
       return 7;
    }
@@ -492,7 +492,7 @@ private int jjMoveStringLiteralDfa8_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(6, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(7, active0);
       return 8;
    }
@@ -517,7 +517,7 @@ private int jjMoveStringLiteralDfa9_0(long old0, long active0){
    if (((active0 &= old0)) == 0L)
       return jjStartNfa_0(7, old0);
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) {
+   catch(EOFException e) {
       jjStopStringLiteralDfa_0(8, active0);
       return 9;
    }
@@ -537,7 +537,7 @@ private int jjStartNfaWithStates_0(int pos, int kind, int state)
    jjmatchedKind = kind;
    jjmatchedPos = pos;
    try { curChar = input_stream.readChar(); }
-   catch(java.io.IOException e) { return pos + 1; }
+   catch(EOFException e) { return pos + 1; }
    return jjMoveNfa_0(state, pos + 1);
 }
 static final long[] jjbitVec0 = {
@@ -1131,7 +1131,7 @@ private int jjMoveNfa_0(int startState, int curPos)
       if ((i = jjnewStateCnt) == (startsAt = 43 - (jjnewStateCnt = startsAt)))
          return curPos;
       try { curChar = input_stream.readChar(); }
-      catch(java.io.IOException e) { return curPos; }
+      catch(EOFException e) { return curPos; }
    }
 }
 private int jjMoveStringLiteralDfa0_1()
@@ -1294,7 +1294,7 @@ private int jjMoveNfa_1(int startState, int curPos)
       if ((i = jjnewStateCnt) == (startsAt = 10 - (jjnewStateCnt = startsAt)))
          return curPos;
       try { curChar = input_stream.readChar(); }
-      catch(java.io.IOException e) { return curPos; }
+      catch(EOFException e) { return curPos; }
    }
 }
 private int jjMoveStringLiteralDfa0_2()
@@ -1427,7 +1427,7 @@ private int jjMoveNfa_2(int startState, int curPos)
       if ((i = jjnewStateCnt) == (startsAt = 7 - (jjnewStateCnt = startsAt)))
          return curPos;
       try { curChar = input_stream.readChar(); }
-      catch(java.io.IOException e) { return curPos; }
+      catch(EOFException e) { return curPos; }
    }
 }
 
@@ -1686,7 +1686,7 @@ public Token getNextToken()
      String error_after = null;
      boolean EOFSeen = false;
      try { input_stream.readChar(); input_stream.backup(1); }
-     catch (java.io.IOException e1) {
+     catch (EOFException e1) {
         EOFSeen = true;
         error_after = curPos <= 1 ? "" : input_stream.GetImage();
         if (curChar == '\n' || curChar == '\r') {
